contracts/SynthSwap.sol

```diff
@@ -0,0 +1,193 @@
+
+import "./interfaces/ISynthSwap.sol";
+import "./interfaces/IERC20.sol";
+import "./interfaces/ISynthetix.sol";
+import "./interfaces/IWETH.sol";
+
+contract SynthSwap is ISynthSwap {
+
+    bytes32 constant SUSD_CURRENCY_KEY = bytes32('sUSD');
+
+    ISynthetix Synthetix;    
+    IERC20 sUSD;
+    IWETH iWETH;
+
+    address volumeRewards;
+    address aggregationRouterV3;
+    address WETH;
+                    
+    event SwapInto(address from, uint amountReceived);
+    event SwapOutOf(address from, uint amountReceived);
+
+    struct OneInchSwapDescription {
+        IERC20 srcToken;
+        IERC20 dstToken;
+        address srcReceiver;
+        address dstReceiver;
+        uint256 amount;
+        uint256 minReturnAmount;
+        uint256 flags;
+        bytes permit;
+    }
+    
+    constructor (
+        address _synthetix, 
+        address _sUSD, 
+        address _volumeRewards, 
+        address _aggregationRouterV3,
+        address _WETH
+    ) {
+        Synthetix = ISynthetix(_synthetix);
+        sUSD = IERC20(_sUSD);
+        volumeRewards = _volumeRewards;
+        aggregationRouterV3 = _aggregationRouterV3;
+        WETH = _WETH;
+    }
+
+    /// @inheritdoc ISynthSwap
+    function swapInto(
+        address inputTokenAddress,
+        uint256 inputTokenAmount,
+        bytes32 destinationSynthCurrencyKey,
+        uint minOut, 
+        bytes calldata _data
+    ) external payable override returns (uint) {
+
+        // Prevent Griefing
+        IERC20 InputERC20 = IERC20(inputTokenAddress);
+        uint tokenBalance = InputERC20.balanceOf(address(this));
+        require(tokenBalance >= inputTokenAmount, "incorrect token balance");
+
+        // Swap InputERC20 with sUSD
+        (uint amountOutputToken, IERC20 outputToken) = swapOnOneInch(minOut, _data);
+        require(outputToken == IERC20(sUSD));
+
+        // Approve the Synthetix router to spend sUSD
+        outputToken.approve(address(Synthetix), amountOutputToken);
+
+        // Swap sUSD with destination synth by providing both the source and destination currency keys
+        uint amountReceived = Synthetix.exchangeWithTrackingForInitiator(
+            SUSD_CURRENCY_KEY, // hardcode source currency key
+            amountOutputToken, // source amount
+            destinationSynthCurrencyKey, // destination currency key
+            volumeRewards, // volume rewards address
+            'KWENTA' // tracking code
+        );
+        
+        emit SwapInto(msg.sender, amountReceived);
+        return amountReceived;
+    }
+
+    /// @inheritdoc ISynthSwap
+    function swapIntoWithETH(
+        bytes32 destinationSynthCurrencyKey,
+        uint minOut, 
+        bytes calldata _data
+    ) external payable override returns (uint) {
+        
+        // Wrap ETH and transfer WETH to this address
+        IWETH(WETH).deposit{value: msg.value}();
+        IERC20 InputERC20 = IERC20(WETH);
+        assert(InputERC20.transfer(address(this), msg.value));
```

Comment:
WETH is already in this contract. This just transfers it back to itself.

https://github.com/WETH10/WETH10/blob/e952d1ec4c149e85d93ed2ce4040ac571ed6bc19/contracts/WETH10.sol#L88